Herança e Interface

ajjaum opened the pull request:
erança, com alterações em Funcionários com a finalização de getters e setters

Interface: Funcionario e IFuncionarios, com a inclusao de promover(String nivelCargo), usando o setNivelCargo.

ajjaum committed:
erança, com alterações em Funcionários com a finalização de getters e setters

Interface: Funcionario e IFuncionarios, com a inclusao de promover(String nivelCargo), usando o setNivelCargo.

pamelasenai commented on objetos/funcionarios/Funcionario.java:
```diff
@@ -47,6 +47,10 @@ public void setSalario(double salario) {
         this.salario = salario;
     }
 
+    public void promover(Professor professor,String nivelCargo) {
```
A classe funcionário se refere aos professores e diretores, então acho que podemos trocar esse parametro professor por Funcionario funcionário, o que você acha?
Sendo assim podemos até retirar as funções setNivelCargo de professor e diretor e trazer pra cá, você concorda?

pamelasenai commented on objetos/funcionarios/IFuncionario.java:
```diff
@@ -1,4 +1,5 @@
 
-public class IFuncionario {
+public interface IFuncionario {
+    void promover(Professor professor, String nivelCargo);
```
A classe funcionário se refere aos professores e diretores, então acho que podemos trocar esse parametro professor por Funcionario funcionário, o que você acha?

pamelasenai reviewed (changes requested):
Ótimo trabalho @ajjaum! 👍 
Deixei apenas umas pequenas sugestões.

ajjaum commented:
Então.

Eu coloquei assim, porque o único que é promovido é professor, de iniciante
para Especialista e deste para avançado. Um último degrau, a meu ver,
poderia ser de de professor avançado para diretor. Como diretor é quem
promove, é estranho ele se "autopromover" ou outro diretor promover um
diretor. O que acha?

Em sáb., 17 de fev. de 2024, 14:43, pamelasenai ***@***.***>
escreveu:

> ***@***.**** requested changes on this pull request.
>
> Ótimo trabalho @ajjaum <https://github.com/ajjaum>! 👍
> Deixei apenas umas pequenas sugestões.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Bruno-FMT/FullStack-Education--P1/pull/9#pullrequestreview-1886727228>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APK4PH652KTHTJN4BJX4Z33YUDT5HAVCNFSM6AAAAABDNOWZS6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTQOBWG4ZDOMRSHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

pamelasenai commented:
O professor tinha falado que um diretor poderia promover o outro diretor e que somente diretores promoveriam professores, também achei estranho, mas ele disse que o diretor não poderia se auto promover.

ajjaum commented:
Entendi!

Vou alterar então!

Obrigado!

Em sáb., 17 de fev. de 2024, 18:58, pamelasenai ***@***.***>
escreveu:

> O professor tinha falado que um diretor poderia promover o outro diretor e
> que somente diretores promoveriam professores, também achei estranho, mas
> ele disse que o diretor não poderia se auto promover.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Bruno-FMT/FullStack-Education--P1/pull/9#issuecomment-1950409078>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APK4PH4RT52CVA5NJPXDCYTYUERZPAVCNFSM6AAAAABDNOWZS6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSNJQGQYDSMBXHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

ajjaum committed:
erança, com alterações em Funcionários com a finalização de getters e setters

Interface: Funcionario e IFuncionarios, com a inclusao de promover(String nivelCargo), usando o setNivelCargo.

ajjaum commented:
O segundo Commit está com a alteração para Funcionario funcionario.

pamelasenai reviewed (approved):
💯 ótima implementação.
